Add Turtlebot3 Burger and Khepera IV Examples (#110)
cyberbotics/webots_ros2: webots_ros2_examples/README.md

```bash
ros2 service call /motor webots_ros2_msgs/SetDifferentialWheelSpeed "{ left_speed: 1.0, right_speed: 0.5 }"
```

#### Display the Sensor Value
The output of the sensor (`/sensor` topic) can be displayed directly using the [ROS2 topic CLI interface](https://index.ros.org/doc/ros2/Tutorials/Introspection-with-command-line-tools)

```bash
ros2 topic echo /sensor
```

#### Troubleshooting

If you see import failures or some indications saying that `WEBOTS_HOME` is incorrectly set, make sure your environment variables are [configured for extern controllers](https://www.cyberbotics.com/doc/guide/running-extern-robot-controllers?version=master#environment-variables).

## TurtleBot3 Burger

![TurtleBot3 Burger in Webots](./assets/turtlebot3_burger.png)

To run TurtleBot3 Burger simulation you can use the universal launcher:
```bash
ros2 launch webots_ros2_core robot_launch.py \
    executable:=webots_differential_drive_node \
    node_parameters:=$(ros2 pkg prefix webots_ros2_examples --share)/resource/turtlebot3_burger.yaml \
    world:=$(ros2 pkg prefix webots_ros2_examples --share)/worlds/turtlebot3_burger_example.wbt
```

> If you have [`turtlebot3`](https://github.com/ROBOTIS-GIT/turtlebot3) package installed you can start `cartographer` as:

Reviewer: ```suggestion
If you have the [`turtlebot3`](https://github.com/ROBOTIS-GIT/turtlebot3) and [cartographer_ros](http://wiki.ros.org/cartographer) packages installed you can start `cartographer` as:
```

Author: But `cartographer_ros` is dependency of `turtlebot3`:
https://github.com/ROBOTIS-GIT/turtlebot3/blob/41e43d4a10aada68b8a7d05675550550091586d6/turtlebot3_cartographer/package.xml#L18

Reviewer: Good point, I probably missed the rosdep step.

```bash
ros2 launch turtlebot3_cartographer cartographer.launch.py \
    use_sim_time:=true

> ```

Reviewer: RVIz is complaining that frame 'map'  does not exist. If we should change something in RVIz it should be documented here?

Maybe this is related to this warning:
``
[occupancy_grid_node-2] [WARN] [1594279529.007253132] [occupancy_grid_node]: submap_slices and last_frame_id is empty
```

Reviewer: Edit: a the second launch it worked fine.

Author: True, it never works from the first launch, I don't know why. It doesn't look like the problem is on our side. I don't know if it is worth investigating it.

Reviewer: Ok, in that case we can just mention that it may not work the first time in parenthesis.

Also, you can start `navigation2` using a custom map:
```bash
ros2 launch turtlebot3_navigation2 navigation2.launch.py \
    use_sim_time:=true \
    map:=$(ros2 pkg prefix webots_ros2_examples --share)/resource/turtlebot3_burger_example_map.yaml

> ```

Reviewer: This seems to fail for me:
```shell
ros2 launch turtlebot3_navigation2 navigation2.launch.py     use_sim_time:=true     map:=$(ros2 pkg prefix webots_ros2_examples --share)/resource/turtlebot3_burger_example_map.yaml
[INFO] [launch]: All log files can be found below /home/david/.ros/log/2020-07-09-09-27-25-415765-david-UX310UQK-87638
[INFO] [launch]: Default logging verbosity is set to INFO
[ERROR] [launch]: Caught exception in launch (see debug for traceback): Caught exception when trying to load file of format [py]: 'TURTLEBOT3_MODEL'
``

Author: It is related to `turtlebot3`, you just need to specify Turtlebot version:
```bash
export TURTLEBOT3_MODEL='burger'
```

I will add it in the documentation.

> Make sure you set initial pose by clicking `2D Pose Estimate` button in RViz or by executing the following command:

Reviewer: ```suggestion
Make sure you set the initial pose by clicking `2D Pose Estimate` button in RViz or by executing the following command:
```

```bash
ros2 topic pub --once /initialpose geometry_msgs/msg/PoseWithCovarianceStamped '{
    "header": { "frame_id": "map" },
    "pose": {
        "pose": {
            "position": { "x": 0.0, "y": 0.0, "z": 0.0 },
            "orientation": { "x": 0.0, "y": 0.0, "z": 0.0, "w": 1.0 }
        }
    }
}'
```

> On the [official TurtleBot3 website](https://emanual.robotis.com/docs/en/platform/turtlebot3/overview/#turtlebot3) you can find more information about [navigation](https://emanual.robotis.com/docs/en/platform/turtlebot3/ros2_navigation2/) and [SLAM](https://emanual.robotis.com/docs/en/platform/turtlebot3/ros2_slam/) configuration.

This will expose necessary topics:
```bash
$ ros2 topic list -t
/scan [sensor_msgs/msg/LaserScan]
/cmd_vel [geometry_msgs/msg/Twist]
/imu [sensor_msgs/msg/Imu]
/joint_states [sensor_msgs/msg/JointState]
/odom [nav_msgs/msg/Odometry]
/robot_description [std_msgs/msg/String]
/tf [tf2_msgs/msg/TFMessage]
/tf_static [tf2_msgs/msg/TFMessage]
```

## Khepera IV

![Khepera IV in Webots](./assets/khepera4.png)

> Similarly to TurtleBot3 Burger you can start Khepera IV robot as well:

Reviewer: ```suggestion
Similarly to the TurtleBot3 Burger you can start the Khepera IV robot as well:
```

```bash
ros2 launch webots_ros2_core robot_launch.py \
    executable:=webots_differential_drive_node \
    node_parameters:=$(ros2 pkg prefix webots_ros2_examples --share)/resource/khepera4.yaml \
    world:=$(ros2 pkg prefix webots_ros2_examples --share)/worlds/khepera4_example.wbt
```

For this robot you should expect the following topics:
```bash
$ ros2 topic list -t

> /camera/camera_info [sensor_msgs/msg/CameraInfo]

Reviewer: I just found that echoing this topic does not print anything:
```shell
ros2 topic echo /camera/camera_info

``

Author: It is latched topic, it should work if you specify QoS:
```
ros2 topic echo --qos-profile services_default --qos-durability transient_local /camera/camera_info
```

```bash
ros2 launch webots_ros2_examples example.launch.py
```
![Thymio II in Webots](assets/ros_example.png)
Webots will start with the Thymio II differential wheel robot and the `example_controller` node will start.
This node acts as a Webots robot controller and publishes the value of the front distance-sensors of the robot on the `/sensor` topic and provides the `/motor` topic which can be used to change the speed of each wheel of the robot.

To quit the simulation and stop the launch file, you simply need to close Webots.

### Interact with the Robot

#### Move the Robot
The `/motor` service can be tested directly using the [ROS2 service CLI interface](https://index.ros.org/doc/ros2/Tutorials/Introspection-with-command-line-tools) to move the robot:

```bash
ros2 service call /motor webots_ros2_msgs/SetDifferentialWheelSpeed "{ left_speed: 1.0, right_speed: 0.5 }"
```

#### Display the Sensor Value
The output of the sensor (`/sensor` topic) can be displayed directly using the [ROS2 topic CLI interface](https://index.ros.org/doc/ros2/Tutorials/Introspection-with-command-line-tools)

```bash
ros2 topic echo /sensor
```

#### Troubleshooting

> If you see import failures or some indications saying that `WEBOTS_HOME` is incorrectly set, make sure your environment variables are [configured for extern controllers](https://www.cyberbotics.com/doc/guide/running-extern-robot-controllers?version=master#environment-variables).

Reviewer 2: I would remove the "?version=master" part from the URL.

## TurtleBot3 Burger

![TurtleBot3 Burger in Webots](./assets/turtlebot3_burger.png)

To run the TurtleBot3 Burger simulation you can use the universal launcher:
```bash
ros2 launch webots_ros2_core robot_launch.py \
    executable:=webots_differential_drive_node \
    node_parameters:=$(ros2 pkg prefix webots_ros2_examples --share)/resource/turtlebot3_burger.yaml \
    world:=$(ros2 pkg prefix webots_ros2_examples --share)/worlds/turtlebot3_burger_example.wbt
```

> If you have [`turtlebot3`](https://github.com/ROBOTIS-GIT/turtlebot3) package installed you can start `cartographer` as (you may need to run the command twice if it fails the first time):

Reviewer 2: If you have **the** ...
Also, I would add a coma after "installed".

```bash
ros2 launch turtlebot3_cartographer cartographer.launch.py \
    use_sim_time:=true
```

Also, you can start `navigation2` using a custom map:
```bash
export TURTLEBOT3_MODEL='burger'
ros2 launch turtlebot3_navigation2 navigation2.launch.py \
    use_sim_time:=true \
    map:=$(ros2 pkg prefix webots_ros2_examples --share)/resource/turtlebot3_burger_example_map.yaml
```

Make sure you set the initial pose by clicking `2D Pose Estimate` button in RViz or by executing the following command:
```bash
ros2 topic pub --once /initialpose geometry_msgs/msg/PoseWithCovarianceStamped '{
    "header": { "frame_id": "map" },
    "pose": {
        "pose": {
            "position": { "x": 0.0, "y": 0.0, "z": 0.0 },
            "orientation": { "x": 0.0, "y": 0.0, "z": 0.0, "w": 1.0 }
        }
    }
}'
```

> On the [official TurtleBot3 website](https://emanual.robotis.com/docs/en/platform/turtlebot3/overview/#turtlebot3) you can find more information about [navigation](https://emanual.robotis.com/docs/en/platform/turtlebot3/ros2_navigation2/) and [SLAM](https://emanual.robotis.com/docs/en/platform/turtlebot3/ros2_slam/) configuration.

> This will expose necessary topics:

Reviewer 2: This will expose **the** necessary topics:
